### PR TITLE
color_format and image_recognizer classes

### PR DESCRIPTION
**Pull request by MarcinTajsner**
- color_format class
- image_recognizer class
- docstrings for both
- new tests
- small changes in resolution class

### CODE REVIEW

File: main_test.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from os import environ
+
+environ["TF_CPP_MIN_LOG_LEVEL"] = "3"
+environ["CUDA_VISIBLE_DEVICES"] = "-1"
+
+from app.image_recognizer import ImageRecognizer # noqa
+
+image_recognizer = ImageRecognizer()
+result = image_recognizer.recognize("tests/test_data/abgr444/picture_nr_5_640x480_abgr444.raw")
+print(result)
```
Comment (KacperSzmitko) [inline main_test.py]: Should this file be in master branch?

File: app/image_recognizer.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        self.resolution_model_img_width = resolution_model_img_width
+        self.resolution_model_img_height = resolution_model_img_height
+        self.color_format_model_img_width = color_format_model_img_width
+        self.color_format_model_img_height = color_format_model_img_height
+
+        if (not exists(self.resolution_model_path)) or (not exists(self.color_format_model_path)):
+            if (self.resolution_model_path == self.DEFAULT_RESOLUTION_MODEL_PATH) and (
+                self.color_format_model_path == self.DEFAULT_COLOR_FORMAT_MODEL_PATH
+            ):
+                self.download_default_models()
```
Comment (KacperSzmitko) [inline app/image_recognizer.py]: What if one path is not default? Then second default model wouldnt be downloaded and exception would be raised

File: app/image_recognizer.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        for resolution in resolutions:
+            print(f"Searching for the best color format for {resolution.width}x{resolution.height} resolution...")
+            color_formats_confidences = self.color_format_finder.find_color_format(raw_img_path, resolution.width)
+            best_color_format = max(color_formats_confidences, key=color_formats_confidences.get)
+            if color_formats_confidences[best_color_format] > best_result.color_format_confidence:
+                best_result.color_format = best_color_format
+                best_result.img_width = resolution.width
+                best_result.img_height = resolution.height
+                best_result.color_format_confidence = color_formats_confidences[best_color_format]
+                best_result.resolution_confidence = resolution.confidence
```
Comment (KacperSzmitko) [inline app/image_recognizer.py:154]: Maybe instead picking result with best confidence, pick the one that occurs most often (counting for example only those with >95 confidence)

File: main_test.py (hunk 1 above)
Comment (MarcinTajsner) [reply main_test.py]: This file will be replaced with proper main by Michał in future, for now it is for demonstration purpose for him.

Comment (KacperSzmitko) [reply main_test.py]: [Here](https://github.com/antmicro-labs/image-recognizer/blob/color_format/run_examples.sh) is a script that would be used to run examples

File: app/image_recognizer.py (hunk 3 above)
Comment (MarcinTajsner) [reply app/image_recognizer.py:154]: I think that picking the best one is better. If detected resolution is incorrect, then color format confidence will be very low anyway, and then the most confident one (with for example 0.2 confidence) may have better chances for being the best, than one that is randomly repeating with even lower confidence score.

File: app/image_recognizer.py (hunk 2 above)
Comment (MarcinTajsner) [reply app/image_recognizer.py]: Yes, that's true, I will change it tomorrow.

File: app/image_recognizer.py (hunk 3 above)
Comment (KacperSzmitko) [reply app/image_recognizer.py:154]: Ye but i think those incorrect resolutions, may still look somehow correct. For example lets assume that we got 3 resolutions and for 2 of them we would get 96 and 97 for rgb. Then for third one lets have 98 for abgr. RGB i thnik in this case would be more accurate.

File: main_test.py (hunk 1 above)
Comment (MarcinTajsner) [reply main_test.py]: Ok, I removed it.

File: app/image_recognizer.py (hunk 2 above)
Comment (MarcinTajsner) [reply app/image_recognizer.py]: Done.

File: app/image_recognizer.py (hunk 3 above)
Comment (MarcinTajsner) [reply app/image_recognizer.py:154]: Incorrect resolution will maybe have 0.2 confidence at best, so this approach would require results to have very low confidence threshold, which could lead to random results.

**Review by KacperSzmitko — APPROVED** [review]

**Pull request merged**